src/Assets/VendorAsset.php

```diff
@@ -0,0 +1,276 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace StellarWP\Assets;
+
+use InvalidArgumentException;
+use LogicException;
+use RuntimeException;
+
+class VendorAsset extends Asset {
+
+	/**
+	 * Whether this is a vendor asset.
+	 *
+	 * @var bool
+	 */
+	protected bool $is_vendor = true;
+
+	/**
+	 * Whether to attempt to load an .asset.php file.
+	 *
+	 * @var bool
+	 */
+	protected bool $use_asset_file = false;
+
+	/**
+	 * VendorAsset constructor.
+	 *
+	 * @param string $slug The asset slug.
+	 * @param string $url  The asset URL.
+	 * @param string $type The asset type.
+	 *
+	 * @throws InvalidArgumentException If the URL is not valid.
+	 */
+	public function __construct( string $slug, string $url, string $type = 'js' ) {
+		$filtered = filter_var( $url, FILTER_VALIDATE_URL );
+		if ( false === $filtered ) {
+			throw new InvalidArgumentException( 'The URL must be a valid URL.' );
+		}
+
+		$this->url  = $filtered;
+		$this->slug = sanitize_key( $slug );
+		$this->type = strtolower( $type );
+	}
+
+	/**
+	 * Registers a vendor asset.
+	 *
+	 * @param string  $slug    The asset slug.
+	 * @param string  $url     The asset file path.
+	 * @param ?string $type    The asset type.
+	 * @param ?string $version The asset version.
+	 *
+	 * @return self
+	 */
+	public static function add( string $slug, string $url, ?string $type = null, $version = null ) {
+		$instance = new self( $slug, $url, $type ?? 'js' );
```

Comment:
`new static`?
Unless we don't want this class extended, then we should make it final 🤔

Reply:
I don't think this particular class would have a need to be extended, but this also shouldn't be `static`. A child class can override this as needed.